### PR TITLE
Wave2 949-985 FE Auto Adjusting Bricks

### PR DESCRIPTION
**Pull request by dpfitzsi**
Series of WAVE2 stories that began with extending the More Info brick to auto-adjust to different layouts and led to a general method for bricks to adjust to column layouts. WAVE2-949, WAVE2-954, WAVE2-885, WAVE2-981, WAVE2-984, WAVE2-985.

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_cta/scss/ucgy_gbl_cta_mini.scss [hunk 1]
```diff
@@ -38,14 +35,12 @@
       -o-transition: opacity 0.5s ease-in-out;
       transition: opacity 0.5s ease-in-out;
 
-      max-width: 266px;
```
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_cta/scss/ucgy_gbl_cta_mini.scss]: Not sure why this change is resulting in too-large Mini CTAs, but they are larger than 266 on desktop...

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_cta/scss/ucgy_gbl_cta_mini.scss [hunk 2]
```diff
@@ -154,52 +153,3 @@
     }
   }
 }
-
-//
-// Adjust layout of Single-CTAs when used in multi-column layouts
```
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_cta/scss/ucgy_gbl_cta_mini.scss]: Thank you for moving this out of the brick CSS!

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_banner/templates/fieldable-panels-pane--ucgy_gbl_banner.tpl.php [hunk 3]
```diff
@@ -99,7 +99,7 @@
                     <div class="col-sm-12">
                         <!-- cta-content div creates a grey bar in tablet view even if there's no content
                         entered by author to display, so we're only displaying this div, if at-least one of fields under cta content is entered by author-->
-                        <?php if(!empty($content['title']['#value']) ||
+                        <?php if( (!empty($content['title']['#value']) && $content['field_disable_heading']['#items'][0]['value'] < 1) ||
```
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_banner/templates/fieldable-panels-pane--ucgy_gbl_banner.tpl.php]: This is still inserting the header if hidden for me (and the resulting grey bar on mobile)?

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+(function ($) {
+
+    function adjustLayout() {
+        // Layout 3
```
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: Can you please document somewhere what/how this whole Layout stuff works? ;)

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        $('.layout-3').find('.col-sm-6').removeClass('col-sm-6').addClass('col-sm-12'); // 2 items
+        $('.layout-3').find('.col-lg-4').removeClass('col-lg-4').addClass('col-lg-12'); // 3 items
+        $('.layout-3').find('.col-lg-3').removeClass('col-lg-3').addClass('col-lg-12'); // 4 items
+        $('.layout-3').find('.no-bottom-margin').removeClass('no-bottom-margin');
+
+        // Layout 4
+        $('.layout-4').find('.col-sm-6').removeClass('col-sm-6').addClass('col-sm-12'); // 2 items
+        $('.layout-4').find('.col-lg-4').removeClass('col-lg-4').addClass('col-lg-12'); // 3 items
+        $('.layout-4').find('.col-lg-3').removeClass('col-lg-3').addClass('col-lg-12'); // 4 items
+        $('.layout-3').find('.no-bottom-margin').removeClass('no-bottom-margin');
```
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: .layout-4 ?

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+        // Layout 12
+        $('.layout-12').find('.col-sm-6').removeClass('col-sm-6').addClass('col-sm-3');
+        $('.layout-12').find('.col-lg-4').removeClass('col-lg-4').addClass('col-md-12');
+        $('.layout-12').find('.col-lg-3').removeClass('col-lg-3').addClass('col-md-12');
+        $('.layout-12').find('.clearfix').removeClass('visible-sm-block');
+    }
+
+    function adjustLayout_matchHeight() {
+        $('.equal-height-container .container').matchHeight();
```
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: I like this automated approach - only for banners in 50/50.

Is this potentially doubling up on the call in theme.js? $('.brick.equal-height [class*="-height"] > [class*="-height"] .brick').matchHeight(); (Really need to eventually move those calls to their respective bricks...)

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_mi/templates/fieldable-panels-pane--ucgy_gbl_mi.tpl.php [hunk 7]
```diff
@@ -1,22 +1,28 @@
 <?php if ($logged_in && $is_admin) { ?><!-- Include file: <?php echo str_replace('-', '=', __FILE__); ?> --><?php } ?>
 <div class="<?php echo $classes; ?>">
-    <div class="row">
+    <div class="row red-back"> <!-- background color -->
```
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_mi/templates/fieldable-panels-pane--ucgy_gbl_mi.tpl.php]: This and other "More Info" brick code are from the mockup and are pre-integration with placeholder to show the back-end devs where things will go. The real "More Info" brick will be merged afterwards.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_mi/templates/fieldable-panels-pane--ucgy_gbl_mi.tpl.php [hunk 8]
```diff
@@ -1,22 +1,28 @@
...
-            <?php if(!empty($content['title']['#value'])||isset($content['field_ucmi_description'])): ?>
-            <div class="row header">
-                <div class="col-sm-12">
-                <?php if(!empty($content['title']['#value'])): ?>
-                    <h2><?php echo $content['title']['#value']; ?></h2>
-                <?php endif; ?>
-                <?php if(isset($content['field_ucmi_description'])): ?>
-                    <?php echo $content['field_ucmi_description'][0]['#markup']; ?>
-                <?php endif; ?>
+            <div class="dark-orange-border-bar"> <!-- border color -->
```
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_mi/templates/fieldable-panels-pane--ucgy_gbl_mi.tpl.php]: More Info mockup.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_text_brick/scss/ucgy_gbl_text_brick.scss [hunk 9]
```diff
@@ -27,4 +27,15 @@
       }
     }
   }
+
+  // don't apply Drupal default 2-column text style in narrow columns
+  .layout-3 .two-col,
```
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_text_brick/scss/ucgy_gbl_text_brick.scss]: Ah! I had put the equivalent markup in layout CSS, but didn't remove it from the brick. Thanks.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_map/templates/fieldable-panels-pane--gbl-map.tpl.php [hunk 10]
```diff
@@ -30,7 +30,7 @@
         <div class="container">
             <div class="row">
                 <div class="col-sm-12">
-                    <h3><?php echo $title; ?></h3>
```
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_map/templates/fieldable-panels-pane--gbl-map.tpl.php]: OK. Reverting to h3.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_mi/scss/ucgy_gbl_mi.scss [hunk 11]
```diff
@@ -1,16 +1,57 @@
...
-  .more-info-item {
-    text-align: left; // Override row-centered
-    vertical-align: top; // Override col-centered
+    .container > div {
+        padding: 15px;
+    }
+
+    .media-body {
+        h3 {
+            margin-top: 10px;
```
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_mi/scss/ucgy_gbl_mi.scss]: "More Info" brick. This was removed post-integration.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_cta/scss/ucgy_gbl_cta_mini.scss (hunk 1 above)
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_cta/scss/ucgy_gbl_cta_mini.scss]: Fixed. Worked for all layouts, but as you said, the size increased to fit the column. Now has default size and then layout adjusts them to fit narrow columns and the 4-column special case.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js (hunk 5 above)
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: .layout-4 are columns that act like .col-sm-4 (33%) columns.

Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: You have a bunch of .layout-4s and one .layout-3 - was just wondering if that last one was a typo?

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js (hunk 6 above)
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: I've made this call specific to .layout-6 (50/50 columns) banners.

There's no doubling up with theme.js that I can tell, because the banner bricks don't have .equal-height classes.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js (hunk 5 above)
Comment (esoltys) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: Ah, gotcha.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_banner/templates/fieldable-panels-pane--ucgy_gbl_banner.tpl.php (hunk 3 above)
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_banner/templates/fieldable-panels-pane--ucgy_gbl_banner.tpl.php]: This is a pre-existing bug and should be raised as a separate ticket - we need a good solution for this as there are other bricks where we are allowing the hiding of 'required' brick headings, and yet they insert code that shows up in the browser.

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js (hunk 4 above)
Comment (TDRoden) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_layout/js/ucgy_core_layout.js]: Documentation being worked on. Thanks Eric. Ultimately have to get in and play!

**Comment by esoltys** [comment]
Reminder to @dpfitzsi that WAVE2-884 More Info brick should be merged **after** this branch, because this branch has only front-end code for More Info while 884 is the fully-integrated version. Thanks.
